05/DB.test.js

```diff
@@ -1 +1,90 @@
-import DB from './DB';
\ No newline at end of file
+import DB from './DB';
+
+describe('data', () => {
+    it('if data.id is 1', () => {
+        const db = new DB();
+        
+        return expect(db.insert({a: 3, b: 4})).resolves.toStrictEqual({a: 3, b: 4, id: 1})
+        
+    })
+    it('if data.id is a number', () => {
+        const db = new DB();
+        
+        return expect(db.insert({a: 3, b: 4, id: "id"})).rejects.toBe('ID can be only number!')
+    })
+    it('rejects if data is not defined', () => {
+        const db = new DB();
+
+        return expect(db.insert()).rejects.toStrictEqual('data is not defined!')
+        
+    })
+    it('if selected id exists', () => {
+        const db = new DB();
+
+        return db.insert({a: 3, b: 4, id: 1})
+        .then(() => expect(db.select(1)).resolves.toStrictEqual({a: 3, b: 4, id: 1}))
+
+    })
+
+    
+
+    it('rejects if data.id is 1', () => {
+        const db = new DB();
+
+        return db.insert({id: 2})
+        .catch((err) => expect(db.select(1)).rejects.toStrictEqual(err))
```

Comment:
Jeśli robisz warunkowy test, a ten taki jest ponieważ wykonujesz porównanie tylko wtedy, gdy będzie błąd (przechwytywany przez `.catch()`) to powinieneś zdefiniować na początku testu ilu porównań się spodziewasz czyli: `expect.assertions(1);`

Wtedy zobaczysz, że ten kod nie przechodzi bo przecież tam jest wszystko ok. Zamiast `catch()` powinno być `then()`.

Całość powinna wyglądać tak:
```
        expect.assertions(1);

        const db = new DB();

        return db.insert({id: 2})
        .then(() => expect(db.select(1)).rejects.toBe('ID not found'))
```